### DISCUSSION TITLE
autho-update

### DISCUSSION THREAD

**Discussion by gmankab** [termux/termux-app#2303] [Ideas]
why don't you make the auto-update functionality like Durov made in telegram?

**Comment by SDRausty** [comment]
>like Durov made in telegram

Can you please describe this to us a little bit more?  Screenshot?

**Reply by HitaloM** [reply]
He's talking about the OTA updates apparently.

![IMG_20210927_075823_802.jpg](https://user-images.githubusercontent.com/40531911/134895943-eb5e5469-1065-4c9f-a1b0-92f36aa3758c.jpg)

**Comment by agnostic-apollo** [comment]
Support for showing a notification when updates are available is already planned for `F-Droid` and `Github` releases but its not high priority. Download and install support can be added too but that will require extra work (support will required anyway when targeting sdk `>=29`). To allow apps to install APKs also requires a special permission to be granted.

**Reply by gmankab** [reply]
I think download and install support is most high priority, because it allows user to fast update app without opening f-droid or github

**Reply by agnostic-apollo** [reply]
F-Droid already sends notifications on updates if battery optimizations are disabled, although the app often has issues. Opening the notification and clicking install is not really that hard. So something that is already working is not high priority, at least for me. And github users are likely pretty low compared to playstore and f-droid users. Decent pull requests are always welcome if someone wants faster feature integration. The `android-10` and `updated-flavor` branches should have related code for installation. For updates, parsing F-Droid and also Github APIs would need to be done, then implementing when to check for updates, then notification creation with android notification channel support, then downloading files from servers itself, then checksum verification, then installation. That's lot of work.
